Eway Bill for Sales Invoice

finbyz opened the pull request:
Reference:- [#15632](https://github.com/frappe/erpnext/pull/15632)
We have modified our PR as suggested by @sagarvora in for [this](https://github.com/frappe/erpnext/pull/15632#issuecomment-428822018) comment from the above PR.

deepeshgarg007 committed:
Accounts Receivable report based on payment terms

deepeshgarg007 committed:
Add columns based on payment terms

pratu16x7 committed:
[setup][init] setup as document records

deepeshgarg007 committed:
Allocation of credit note amount and pdc in payment terms

pratu16x7 committed:
[setup] fetch setup config

pratu16x7 committed:
[setup] set the default fiscal year that was made

deepeshgarg007 committed:
Changed columns and added currency conditions

pratu16x7 committed:
[setup] don't premake company fixtures

pratu16x7 committed:
[setup] bundle up fixtures and setup docs

rohitwaghchaure committed:
[Fix] TDS is applying on the tax

pratu16x7 committed:
[setup][fix] get newly created root Territory for children

pratu16x7 committed:
[setup][fix] company fixtures

pratu16x7 committed:
[setup] add market segments and sales stages

pratu16x7 committed:
[setup] add get_country_and_charts()

pratu16x7 committed:
[setup] separate out setup_company

pratu16x7 committed:
Rename 'Chat' to 'Gitter Chat'

pratu16x7 committed:
[fix] set default_bank_account in company

pratu16x7 committed:
[fix] >.< call setup_taxes

pratu16x7 committed:
[setup] remove redundant code

- [fix] domain setup from dictionary

pratu16x7 committed:
[fix] setup chart of accounts

deepeshgarg007 committed:
List view currency bug fix

rohitwaghchaure committed:
[Fix] In project, tax amount is added in the total billed amount(sales) field

netchampfaris committed:
fix(setup-wizard): Validate abbr length before switching to next slide

deepeshgarg007 committed:
Credit note fix

deepeshgarg007 committed:
Refactored accounts receivable report for payment terms

deepeshgarg007 committed:
Removed column for pdc date

deepeshgarg007 committed:
Merge branch 'hotfix' into list_view

nabinhait committed:
Merge pull request #16144 from deepeshgarg007/list_view

[Fix]List view currency bug fix

nabinhait committed:
Merge pull request #16146 from rohitwaghchaure/remove_tax_amt_from_the_billed_amt_in_project

[Fix] In project, tax amount is added in the total billed amount(sales) field

nabinhait committed:
Merge pull request #16163 from netchampfaris/hotfix-setupwizard-abbr

fix(setup-wizard): Validate abbr length before switching to next slide

SaiFi0102 committed:
Added root password to .travis.yml

SaiFi0102 committed:
Rearranged arguments

frappe-bot committed:
Merge branch 'hotfix'

frappe-bot committed:
bumped to version 10.1.75

surajshetty3416 committed:
Fix user permission checks

pratu16x7 committed:
[setup] remove unused imports

pratu16x7 committed:
[fixtures] re-add market_segments and sales_stages utils

pratu16x7 committed:
[setup][minor] remove unused import

pratu16x7 committed:
[setup] fix install_defaults

pratu16x7 committed:
[setup] add defaults stage

finbyztech committed:
Added Transporter Info Section in Sales Invoice

finbyztech committed:
Added Transporter Info Section in Sales Invoice

finbyztech committed:
Modified Eway Bill for Sales Invoice

finbyztech committed:
Modified Eway Bill for Sales Invoice

finbyz commented on erpnext/regional/india/setup.py:
```diff
@@ -201,6 +201,35 @@ def make_custom_fields(update=True):
+			insert_after='mode_of_transport', options='\nRegular\nOver Dimensional Cargo (ODC)', default='Regular',
+			depends_on='eval:(doc.mode_of_transport === "Road")', translatable=1, print_hide=1),
+	]
+
```
Added Transporter Info section in Sales Invoice

rmehta commented:
Can you please add a screenshot of the e-way bill? And maybe add simple test to check the output of the bill?

codingCoffee committed:
Merge pull request #16190 from SaiFi0102/Travis-Fix

[Staging] Travis Fix: bench reinstall allow setting MariaDB root user/pass #6599

netchampfaris committed:
fix(product-search): Include variants in product search

pratu16x7 committed:
[setup] domain not in array

pratu16x7 committed:
Merge pull request #15964 from pratu16x7/staging-fixes-setup-refactor

[setup][init] setup as document records

saurabh6790 committed:
bumped to version 11.0.3-beta.30

Zlash65 committed:
make quality inspection in query configurable

Zlash65 committed:
fix fetch material request items logic

netchampfaris committed:
fix(pos-profile): Cleanup Form for POS Profile (#16208)

- Remove pos_profile_name field and use `name` as Prompt
- Reorganize form and label sections
- Remove `apply_discount` which was not used

nabinhait committed:
fix(variant): Show attribute values on single variant creation (#16204)

sagarvora commented:
@finbyz Thanks for making the necessary changes.

I've planned a feature development to generate a JSON file from DN/SINV itself based on the spec provided on eWaybill website, but its not on priority right now.

Can you please add a screenshot for the Transporter Info section in Sales Invoice (after patch is applied)?

Also a good idea might be ensuring the patch (https://github.com/frappe/erpnext/blob/staging-fixes/erpnext/patches/v11_0/ewaybill_fields_gst_india.py)  runs for those already on v11, but I don't know how that can be implemented.

netchampfaris committed:
Merge pull request #16201 from netchampfaris/fix-product-search

fix(product-search): Include variants in product search

deepeshgarg007 committed:
Analytics report graph fix

Zlash65 committed:
fix check for leave on holiday for half day

rohitwaghchaure committed:
[Fix] Get items from material request not working for PO

rohitwaghchaure committed:
Merge pull request #16211 from Zlash65/fix-pp

[Minor] Get items for Material Requests fix

rohitwaghchaure committed:
Merge pull request #16219 from Zlash65/fix-holiday-leave

[Minor] Fix Leave Application as half day on Holiday validation

rohitwaghchaure committed:
Merge pull request #16207 from Zlash65/fix-qi

[Minor] Make Quality Inspection in query configurable

shreyashah115 committed:
fix: Remove date_of_joining from field list

deepeshgarg007 committed:
Added test case for Accounts receivable report based on payment terms

shreyashah115 committed:
fix: remove paid_amount and modify query

deepeshgarg007 committed:
breaked up code into multiple functions

rohitwaghchaure committed:
Merge pull request #16228 from shreyashah115/fix-student-report

Fix Student Fee Collection report

rohitwaghchaure committed:
Merge pull request #16226 from shreyashah115/fix-list

Fix employee separation list

rohitwaghchaure committed:
Merge pull request #16225 from rohitwaghchaure/not_able_to_make_po_from_mr

[Fix] Get items from material request not working for PO

Alchez committed:
fix(customer): Fix Total Unpaid amount in Customer dashboard (#16234)

surajshetty3416 committed:
Fix: currency symbol issue in delivery note list (#16233)

surajshetty3416 committed:
Merge branch 'staging-fixes' into permission-fixes

nabinhait committed:
Merge pull request #16218 from deepeshgarg007/analytics-graph

[Bug-fix] Analytics report graph skipping quarter values bug fix

deepeshgarg007 committed:
Added temp variables for outstanding, credit_note_amt instead of function calling

deepeshgarg007 committed:
Rebase using staging-fixes and resolved conflicts

surajshetty3416 committed:
Commonify code
- use get_permitted_documents function to avoid code redundancy

finbyz commented:
@sagarvora Here is the screenshot of Transporter Info Section after patch is applied.

![screenshot 6](https://user-images.githubusercontent.com/18363620/50270753-a6961e80-0458-11e9-8a71-4caf9a3f9a34.png)

And secondly, we have already applied the patch on ([https://github.com/frappe/erpnext/blob/staging-fixes/erpnext/patches/v11_0/ewaybill_fields_gst_india.py](https://github.com/frappe/erpnext/blob/staging-fixes/erpnext/patches/v11_0/ewaybill_fields_gst_india.py)) that you provided.

shreyashah115 committed:
fix: Check if items exist

nabinhait committed:
Update accounts_receivable.py

nabinhait committed:
Merge pull request #15973 from deepeshgarg007/accounts-receivable

Accounts receivable report based on payment terms

chdecultot committed:
Filter cancelled and draft payments

nabinhait committed:
Merge pull request #16249 from shreyashah115/error-fix-2

Purchase order validation sql error

frappe-bot committed:
bumped to version 11.0.3-beta.31

frappe-bot committed:
Merge branch 'staging-fixes' into staging

surajshetty3416 committed:
Fix daily_work_summary_settings migration patch

surajshetty3416 committed:
Merge branch 'staging-fixes' into permission-fixes

netchampfaris committed:
Merge pull request #16269 from surajshetty3416/fix-daily-work-summary-patch

fix: daily_work_summary_settings migration patch

deepeshgarg007 committed:
Removed subscriber option form module page (#16270)

rohitwaghchaure committed:
[Fix] Warehouse wise item balance age and value report not working (#16258)

barredterra committed:
Update README.md (#16224)

chdecultot committed:
[Fix] FEC: Correction of the export function (#16252)

* Correction of the export function

* Improve JS code

* Codacy corrections

* Codacy corrections

shreyashah115 committed:
fix: Check if items exist (#16248)

chdecultot committed:
Safe decode project name for languages with special characters (#16254)

shreyashah115 committed:
fix: add currency to options (#16199)

Mangesh-Khairnar committed:
[Enhancement] Pending SO Items for Purchase Request Report  (#16188)

* deleted old purchase request query report

* [Enhancement] pending so items for purchase request report refactor

* comments and format fix

* test and minor fixes

* test fixes

rohitwaghchaure committed:
Rename BOM and Work order fields (#16209)

* Rename BOM and Work order fields

* Rearranged BOM fields

chdecultot committed:
[Fix] Pos bugs (#16256)

* Remove hard-coded values

* Fix change pos profile + add campaign

* Test case for coveralls

* Remove unnecessary method

Alchez committed:
fix(work_order): Set operations for Work Order when created from MREQ (#16154)

KanchanChauhan committed:
Set ignore_user_permission 1 for Company field in Item Default (#16167)

navdeepghai committed:
Fix the disappears of image bug after uploading and saving the employee (#16217)

* Fix image disappears of image after uploading and saving the employee

* Fix the codacy issue

vjFaLk committed:
fix: set fieldname for frappe prompt (#16198)

nabinhait committed:
feat(quality): Check quality status before receipt/delivery (#16169)

rohitwaghchaure committed:
[Fix] While making sales invoice from delivery note, system not remove the returned qty (#16141)

rohitwaghchaure committed:
[Fix] Salary structure not saving (#16275)

surajshetty3416 committed:
Modify bench_init.sh
- to point frappe branch in my repo with
changes to check if any test fails due to the changes

surajshetty3416 committed:
Merge branch 'permission-fixes' of https://github.com/surajshetty3416/erpnext into permission-fixes

surajshetty3416 committed:
Skip some codacy warnings

rohitwaghchaure committed:
Merge pull request #16016 from rohitwaghchaure/fixed_tax_withholding_issue

[Fix] Tax Withholding, TDS is applying on the tax

surajshetty3416 committed:
Revert "Modify bench_init.sh"

Point back to original repo

This reverts commit d3530125dd73a0a782758fbfedf73690e3f728f5.

rohitwaghchaure committed:
[Fix] Patch rename_bom_wo_fields

chdecultot committed:
Test cases

chdecultot committed:
Remove ununsed variable

chdecultot committed:
Codacy correction

rohitwaghchaure committed:
Merge pull request #16257 from surajshetty3416/permission-fixes

Fix user permission checks

chdecultot committed:
Ignore pricing rule for Travis

rohitwaghchaure committed:
Merge pull request #16286 from rohitwaghchaure/fixed_patch_rename_bom_wo_fields

[Fix] Patch rename_bom_wo_fields

frappe-bot committed:
Merge branch 'hotfix'

frappe-bot committed:
bumped to version 10.1.76

frappe-bot committed:
Merge branch 'master' into staging-fixes

frappe-bot committed:
Merge branch 'staging-fixes' into staging

frappe-bot committed:
bumped to version 11.0.3-beta.32

chdecultot committed:
Corrections for Travis

chdecultot committed:
Understand Travis

chdecultot committed:
Remove pricing rule from items

chdecultot committed:
Add own records

chdecultot committed:
Cancel existing payment entries for data integrity in test

chdecultot committed:
Test travis

chdecultot committed:
Cleanup journal entries

chdecultot committed:
Correction for Travis

rohitwaghchaure committed:
[Fix] Duplicate fieldnames operation and allow_alternative_item in BOM Item

sahil28297 committed:
fix: Checks for status in employee and for date range in upload attendance. (#16279)

* fix: checks for status in employee and for date range in upload attendance.

1. Earlier the status of an amployee can be set to 'Left' even though other employees are reporting to this employee, added checks to prevent this behaviour.
2. Earlier the range of dates added to the template were not checked for date of joining and relieving date. Now the range of dates added in the template are between the date of joining and relieving date.

* remove "import erpnext"

* fix: replace frappe.db.sql with frappe.db.get_all

* fix: refactored using list comprehension

* fix: query refactoring

* fix: combining list comprehensions

* travis debugging

* fix: doc.save

rohitwaghchaure committed:
Merge pull request #16306 from rohitwaghchaure/removed_duplicate_fields

[Fix] Duplicate fieldnames operation and allow_alternative_item in BOM Item

nabinhait committed:
Merge pull request #16253 from chdecultot/sales_summary_correction

[Fix] Filter cancelled and draft payments in sales payment summary

rohitwaghchaure committed:
[Fix] Not able to delete department

rohitwaghchaure committed:
Added test cases

chdecultot committed:
Fix #15917 (#16311)

rohitwaghchaure committed:
Merge pull request #16312 from rohitwaghchaure/fix_not_able_to_delete_department

[Fix] Not able to delete department

sagarvora committed:
fix: leave application and timesheet tests

sagarvora committed:
Merge pull request #16323 from sagarvora/fix_tests

fix: leave application and timesheet tests

Commit:
fix: remove print statement from test case

sagarvora committed:
Merge pull request #16338 from sahil28297/remove_print_from_test_case

fix: Remove print statement from test case

finbyztech committed:
Merge branch 'staging-fixes' of https://github.com/finbyz/erpnext into staging-fixes

test2a commented:
@finbyz any idea if and when the latest update on eway bill would be implemented?
